TicTacToeApp/ViewController.swift

```diff
@@ -1,152 +1,99 @@
     super.viewDidLoad()
-    updateGamePrompt()
+    gameBoard = BoardButtons(buttons: boardButtons)
+    gamePrompt = GamePrompt(prompt: prompt)
+    confirmationAlert = createAlert("Are you sure?", message: "Current game progress will be lost.")
+    gamePrompt?.updateFor(gameState)
     resetButton.enabled = false
   }
 
-  // Rename to resetGame
-  @IBAction public func triggerGameReset() {
-    resetGameWithConfirmation()
-  }
-
   @IBAction public func makeMove(button: UIButton) {
-    updateGameBoardForPlayerMove(button.tag, playerMark: game.getCurrentPlayer())
-    game.makeMove(button.tag)
-    updateGamePrompt()
+    gameBoard?.markButton(gameState.getCurrentPlayer(), spaceId: button.tag)
 
-    if (!resetButton.enabled) {
-      resetButton.enabled = true
-    }
+    gameState.makeMove(button.tag)
+    gamePrompt?.updateFor(gameState)
+    resetButton.enabled = true
 
-    if game.playerWonLastTurn(game.getInactivePlayer()) || game.isADraw() {
-      disableBoardButtons()
-    } else if (currentMode == GameMode.HumanVsComputer && game.getCurrentPlayer() == PlayerMark.O) {
-      makeMove(getButtonByTag(ComputerPlayer().makeMove(game))!)
+    if gameIsOver(gameState) {
+      gameBoard?.disableInput()
+    } else if isComputersTurn(gameState) {
+      makeMove(getCorrespondingButton(ComputerPlayer().makeMove(gameState))!)
     }
   }
 
-  @IBAction public func triggerGameModeChange(button: UIButton) {
-    changeGameMode(getModeFrom(button))
-  }
-
-  private func changeGameMode(mode: GameMode) {
-    if (mode == GameMode.HumanVsHuman) {
-      currentMode = GameMode.HumanVsHuman
-    } else {
-      currentMode = GameMode.HumanVsComputer
-    }
-  }
-
-  public func resetGameWithConfirmation(confirmationDialogue: UIAlertController =
-    UIAlertController(title: "Are you sure?",
-      message: "Current game progress will be lost.",
-      preferredStyle: UIAlertControllerStyle.Alert)) {
-        confirmUserAction(confirmationDialogue, okAction: { self.resetGame() })
-  }
-
-  public func resetGame() {
-    game = Game()
-    clearBoard()
-    enableBoardButtons()
-    enableResetButton()
-    updateGamePrompt()
-  }
-
-  private func enableResetButton() {
-    resetButton.enabled = false
-  }
-
-  private func confirmUserAction(confirmationDialogue: UIAlertController, okAction: () -> Void ) {
-    addAlertAction(confirmationDialogue, title: "OK", action: {(action: UIAlertAction!) in
-      okAction()
-    })
-    addAlertAction(confirmationDialogue, title: "Cancel", action: {(_) in })
-
-    presentViewController(confirmationDialogue, animated: true, completion: nil)
-  }
-
-  private func addAlertAction(alertController: UIAlertController, title: String, action: (UIAlertAction) -> Void) {
-    alertController.addAction(UIAlertAction(title: title, style: .Default, handler: action))
-  }
-
-  private func updateGamePrompt() {
-    infoLabel.text = gamePrompt.promptFor(game)
+  private func gameIsOver(game: Game) -> Bool {
```

Comment:
This seems to be an SRP violation. The `ViewController` is knowing details of the rules of the game.